src/Beacon.jsx

```diff
@@ -83,13 +97,21 @@ export class Beacon extends Component {
   }
 
   shouldComponentUpdate(nextProps, nextState) {
-    if (nextState.tooltip) {
-      // Always update if the tooltip is visible
+    if (Object.keys(this.props).some(key => this.props[key] !== nextProps[key])) {
+      // Always update if any prop has changed
       return true;
-    } else if (Object.keys(this.props).some(key => this.props[key] !== nextProps[key])) {
-      // Update if any prop has changed
+    } else if (this.state.tooltip !== nextState.tooltip) {
       return true;
-    } else if (this.state.inactive !== nextState.inactive || this.state.hashCheck !== this.checkHashStatus()) {
+    } else if (
+      nextState.tooltip &&
+      TOOLTIP_STATE_VARIABLES.some(key => this.state[key] !== nextState[key])
+    ) {
```

Comment:
I had to optimize for `tooltip === true` as well since I was getting superfluous renders there as well (mainly when the hash status changes, even though this doesn't affect how the tooltip is rendered).